.github/workflows/test-and-format.yml

```diff
@@ -0,0 +1,50 @@
+name: Test & Format
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  id-token: write
+  contents: read
+  checks: write
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    # strategy:
+    #   matrix:
+    #     dotnet-version: [ '3.1.x', '6.0.x' ]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v3
+        # with:
+        #   dotnet-version: ${{ matrix.dotnet-version }}
+      # You can test your matrix by printing the current dotnet version
+      - name: Display dotnet version
+        run: dotnet --version
+      - name: Install dependencies
+        run: dotnet restore --locked-mode
```

Comment:
This repo doesn't use lock files, or is this in preparation to do it in the future?

Reply:
Since the build works regardless I thought i could just as well use it?

Reply:
Yeah that's fine. Just wondering if you also meant to introduce them.